### ISSUE TITLE
Build fails when setting skipFeatureDeployment to true

### ISSUE THREAD

**Issue by barisbikmaz** [SharePoint/sp-dev-docs#7386]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [X] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- SPFx version: 1.12.1
- Node.js version: 14.17.0

### Describe the bug / error

I have a solution which I want to install tenant-wide. Therefore I have set "skipFeatureDeployment" property to "true".
When I run the command "gulp package-solution --ship" I get a warning 

**Admins can make this solution available to all sites immediately, but the solution also contains feature.xml elements for provisioning. Feature.xml elements are not automatically applied unless the solution is explicitly installed on a site.**

which results in an npm error and causes our CI pipeline to fail.

### Steps to reproduce

1. Goto package-solution.json
2. Set skipFeatureDeployment to "true"
3. run "gulp package-solution --ship"

My package-solution.json

```
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/package-solution.schema.json",
  "solution": {
    "name": "My Solution Name",
    "id": "13aa1234-5555-4455-aaaa-1a2eab3f4567",
    "iconPath": "assets/icons/app-icon.png",
    "version": "1.0.0.0",
    "includeClientSideAssets": true,
    "skipFeatureDeployment": true,
    "features": [
      {
        "title": "My cool feature",
        "description": "Deploys a command button for document libraries.",
        "id": "2b8fcbc1-5aa7-4ac7-8e5f-74378d2119fb",
        "version": "1.0.0.0",
        "assets": {
          "elementManifests": [
            "elements.xml",
            "clientsideinstance.xml"
          ]
        }
      }
    ],
    "developer": {
      ....
    }
  },
  "paths": {
    "zippedPackage": "solution/my-solution.sppkg"
  }
}
```

package.json

 ```
 "dependencies": {
    "@microsoft/decorators": "1.12.1",
    "@microsoft/sp-core-library": "1.12.1",
    "@microsoft/sp-dialog": "1.12.1",
    "@microsoft/sp-listview-extensibility": "1.12.1",
    "@pnp/sp": "^2.9.0",
    "@types/webpack-env": "^1.16.2"
  },
  "devDependencies": {
    "@microsoft/rush-stack-compiler-3.7": "0.6.47",
    "@microsoft/sp-build-web": "1.12.1",
    "@microsoft/sp-module-interfaces": "1.12.1",
    "@microsoft/sp-tslint-rules": "1.12.1",
    "@microsoft/sp-webpart-workbench": "1.12.1",
    "@types/chai": "4.2.22",
    "@types/jest": "24.0.24",
    "@types/mocha": "9.0.0",
    "@voitanos/jest-preset-spfx-react16": "^1.5.0",
    "ajv": "~8.6.3",
    "gulp": "^4.0.2",
    "jest": "24.9.0",
    "ts-jest": "24.2.0",
    "typescript": "3.7.7"
  }
```

### Expected behavior

Setting skipFeatureDeployment property to true should not fail build.

**Comment by barisbikmaz** [comment]
@AJIXuMuK Yes my CI pipeline fails because of this warning. As you can see in my previous comment npm sends an error code 1. 

At the moment I have added a build supression to gulpfile.js:
  -  build.addSuppression(/Admins can make this solution available to all sites immediately.*/);

**Comment by AJIXuMuK** [comment]
@barisbikmaz - the promised changes were included in the 1.14 beta.
